packages/dendron-next-server/pages/vscode/calendar-view.tsx

```diff
@@ -0,0 +1,187 @@
+import {
+  createLogger,
+  engineSlice,
+  ideHooks,
+  postVSCodeMessage,
+} from "@dendronhq/common-frontend";
+import {
+  DMessageSource,
+  CalendarViewMessageType,
+  DNodeUtils,
+  NoteUtils,
+  NoteProps,
+} from "@dendronhq/common-all";
+import type { Moment } from "moment";
+import moment from "moment";
```

Comment:
unless ant forces you into moment, lets use `luxon` for dealing with times. see [here](https://github.com/dendronhq/dendron-site/blob/dev/vault/dendron.dev.style.md#L26:L26). our current dependencies on moment should actually be removed since its not being used anywhere else

Reply:
The code to use `luxon` would look like:

```ts
import { DateTime } from "luxon";
import dayjsGenerateConfig from "rc-picker/lib/generate/luxon"; // this is currently not available
import generateCalendar from "antd/es/calendar/generateCalendar";
import "antd/es/calendar/style";

const Calendar = generateCalendar<DateTime>(dayjsGenerateConfig);
```

But `rc-picker/lib/generate/luxon` is currently not available and has been brought to [attention](https://github.com/ant-design/ant-design/issues/29710) and a [PR](https://github.com/react-component/picker/pull/230) has been created.

Reply:
For reference the entailed changes.
```diff
diff --git a/packages/dendron-next-server/pages/vscode/calendar-view.tsx b/packages/dendron-next-server/pages/vscode/calendar-view.tsx
index 885bbf55..dccb8928 100644
--- a/packages/dendron-next-server/pages/vscode/calendar-view.tsx
+++ b/packages/dendron-next-server/pages/vscode/calendar-view.tsx
@@ -13,14 +13,21 @@ import {
 } from "@dendronhq/common-all";
 import type { Moment } from "moment";
 import moment from "moment";
-import { Calendar, Badge, Space } from "antd";
+import { Badge, Space } from "antd";
 import { blue } from "@ant-design/colors";
 import type { CalendarProps } from "antd";
 import _ from "lodash";
 import React, { useState, useMemo, useCallback, useEffect } from "react";
 import { DendronProps } from "../../lib/types";
 
-type AntdCalendarProps = CalendarProps<Moment>;
+import { DateTime } from "luxon";
+import dayjsGenerateConfig from "rc-picker/lib/generate/dayjs";
+import generateCalendar from "antd/es/calendar/generateCalendar";
+import "antd/es/calendar/style";
+
+const Calendar = generateCalendar<DateTime>(dayjsGenerateConfig);
+
+type AntdCalendarProps = CalendarProps<DateTime>;
 
 function getMaybeDatePortion({ fname }: NoteProps, journalName: string) {
   const journalIndex = fname.indexOf(journalName);
@@ -88,14 +95,14 @@ function CalendarView({ engine, ide }: DendronProps) {
       );
 
       return maybeDatePortion && _.first(groupedDailyNotes[maybeDatePortion])
-        ? moment(maybeDatePortion, defaultJournalDateFormat)
+        ? DateTime.fromFormat(maybeDatePortion, defaultJournalDateFormat)
         : undefined;
     }
   }, [noteActive, groupedDailyNotes]);
 
   const getDateKey = useCallback(
-    (date: Moment) => {
-      return date.format(
+    (date: DateTime) => {
+      return date.toFormat(
         currentMode === "month" ? defaultJournalDateFormat : "y.MM" // TODO compute format for currentMode="year"
       );
     },
@@ -132,7 +139,9 @@ function CalendarView({ engine, ide }: DendronProps) {
     [setCurrentMode]
   );
 
-  const dateCellRender: AntdCalendarProps["dateCellRender"] = useCallback(
+  const dateCellRender = useCallback<
+    Exclude<AntdCalendarProps["dateCellRender"], undefined>
+  >(
     (date) => {
       const dateKey = getDateKey(date);
       const dailyNotes = groupedDailyNotes[dateKey] ?? [];
```